doc/guides/nest_server.rst

```diff
@@ -1,37 +1,566 @@
+    response = nsc.exec_script("print('Hello world!')")
+    print(response['stdout'])                                             # 'Hello world!'
+    response = nsc.exec_script("models=nest.Models()", 'models')
+    models = response['data']
+    print(models)                                                         # a list of models
+
+NEST Server Client is able to reads script from file with the ``from_file`` method and then executes it on server side.
+The code can be taken from typical Python script (e.g. ``simulation_script.py`` in left column in Python section).
+
+.. code-block:: Python
+
+    from NESTServerClient import NESTServerClient
+    nsc = NESTServerClient()
+    response = nsc.from_file('simulation_script.py', 'n_events')
+    n_events = response['data']
+    print('Number of events:', n_events)
 
 
 Limitations and security implications
 -------------------------------------
 
-   
+The code execute function of Python follows with a security risk on server side.
+An unauthorized access with Python script could corrupt the system.
+
+To ensure to keep the server secure, we created a trusted environment with limitation.
+The limitation helps the server to control module imports.
+Moreover RestrictedPython perserves a high security standard on server side.
+
+.. note::
+
+    RESTful API is not affected from the restricted environment.
+
+**Run with Python modules**
+
+You are not able to import any modules in code execution.
+With starting NEST Server, all registered modules are imported.
+Here, we show codes to register more modules for code execution in NEST Server.
+As an example, we want to run script with numpy.
+
+.. code-block:: sh
+
+    export NEST_SERVER_MODULES=nest,numpy
+    nest-server start
+
+Then, execute a code to create an array of NumPy and return it as list.
+
+.. code-block:: Python
+
+    from NESTServerClient import NESTServerClient
+    nsc = NESTServerClient()
+    response = nsc.exec_script("a = numpy.arange(10)", 'a')
+    print(response['data'][::2])                                          # [0, 2, 4, 6, 8]
+
+**RestrictedPython**
+
+RestictedPython is a tool helps to define a trusted environment and to execute untrusted code inside of it.
+By default, NEST Server runs with restricted environment (with RestrictedPython).
+However, some code might not work in restriction.
+We also implemented an option to turn off the trusted environment.
+
+.. warning::
+
+    In case without RestrictedPython you should be aware of security risk of the server.
+
+We show steps how to run NEST Server without restricted environment.
+
+.. code-block:: sh
+
+    export NEST_SERVER_RESTRICTION_OFF=true
+    nest-server start
+
+.. code-block:: Python
+
+    from NESTServerClient import NESTServerClient
+    nsc = NESTServerClient()
+    response = nsc.exec_script("print(nest.__version__)")
+    print(response['stdout'])                                             # 'HEAD@ef42c5f2f'
+
 
+Tools that use NEST Server
```

Comment:
I would add a sentence prior to the list, e.g., "A variety of tools use NEST Server, including the"